scripts/o3de/o3de/project_properties.py

```diff
@@ -43,6 +43,19 @@ def _edit_gem_names(proj_json: dict,
         add_list = new_gem_names.split() if isinstance(new_gem_names, str) else new_gem_names
         if is_optional_gem:
             add_list = [dict(name=gem_name, optional=True) for gem_name in add_list]
+
+        def is_version_of_gem(candidate: str or dict, gem_name) -> bool:
+            candidate_gem_name = candidate if isinstance(candidate, str) else candidate.get('name')
+            candidate_gem_name_only, _ = utils.get_object_name_and_optional_version_specifier(candidate_gem_name)
+            return candidate_gem_name_only == gem_name
+
+        # remove any versions of the existing gem first
+        if proj_json.get('gem_names',[]):
+            for new_gem_name in add_list:
+                new_gem_name = new_gem_name if isinstance(new_gem_name,str) else new_gem_name['name']
+                gem_name_only, _ = utils.get_object_name_and_optional_version_specifier(new_gem_name)
+                proj_json['gem_names'] = [gem for gem in proj_json['gem_names'] if not is_version_of_gem(gem, gem_name_only)]
```

Comment:
Doesn't the delete_gem_names code need this logic as well?

If the project.json contained the following entry
```json
{
    "gem_names": [ "gemA=1.0.0" ]
}
```

And the user requested to remove gem "gemA", then "gemA=1.0.0" would still exist.

Reply:
That's probably a good idea. Currently you have to specify the exact gem name with specifier or it won't delete it, but I bet if a user tries to remove a gem and doesn't provide the specifier, we can assume they want to remove any version of the gem.